src/modules/components/proposal/proposalVoting/proposalVotingBodyContent/proposalVotingBodyContent.test.tsx

```diff
@@ -151,7 +123,42 @@ describe('<ProposalVotingBodyContent /> component', () => {
         render(createTestComponent({ bodyId }, contextValues));
 
         expect(screen.queryByRole('button', { name: 'All bodies' })).not.toBeInTheDocument();
-        expect(screen.queryByText('Test Stage')).not.toBeInTheDocument();
+    });
+
+    it('does not render back button when bodyList has one or fewer elements', () => {
+        const bodyId = 'body1';
+        const activeBody = 'body1';
+        const contextValues = {
+            activeBody: activeBody,
+        };
+
+        render(createTestComponent({ bodyId }, contextValues));
+
+        expect(screen.queryByRole('button', { name: 'All bodies' })).not.toBeInTheDocument();
+    });
+
+    it('renders back button when bodyList has more than one element', () => {
+        const bodyId = 'body1';
+        const activeBody = 'body1';
+        const contextValues = {
+            bodyList: ['body1', 'body2'],
+            activeBody: activeBody,
+        };
+
+        render(createTestComponent({ bodyId }, contextValues));
+
+        expect(screen.getByRole('button', { name: 'All bodies' })).toBeInTheDocument();
+    });
+
+    it('renders the body name when provided', () => {
+        const bodyId = 'body1';
+        const activeBody = 'body1';
+        const name = 'Active test';
+        const contextValues = { activeBody, bodyList: undefined };
+
+        render(createTestComponent({ bodyId, name }, contextValues));
+
+        expect(screen.getByText(name)).toBeInTheDocument();
```

Comment:
just grouping these tests for 'back button' and removing check for name as is now external -- see third assertion of block diff.